clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -99,8 +99,8 @@
     private final ConcurrentLinkedQueue<CompletedFetch> completedFetches;
     private final Deserializer<K> keyDeserializer;
     private final Deserializer<V> valueDeserializer;
-
     private PartitionRecords nextInLineRecords = null;
+    private ExceptionMetadata nextInLineExceptionMetadata = null;
```

Comment:
But I think we still need to cache these to handle the exception thrown from `Fetcher.fetchRecords()`, which in turn comes from `Fetcher.maybeEnsureValid()`. What do you think?